tests/Builder/Type/QueryObjectTest.php

```diff
@@ -51,19 +66,23 @@ public function provideQueryObjectValue(): Generator
         yield 'string' => [['foo' => 'bar']];
         yield 'bool' => [['foo' => true]];
         yield 'null' => [['foo' => null]];
+        yield 'decimal128' => [['foo' => new Decimal128('1.1')]];
+        yield 'int64' => [[1 => new Int64(1)]];
         yield 'regex' => [['foo' => new Regex('foo')]];
         yield 'object' => [['foo' => (object) ['bar' => 'baz']]];
         yield 'list' => [['foo' => ['bar', 'baz']]];
         yield 'operator as array' => [['foo' => ['$eq' => 1]]];
         yield 'operator as object' => [['foo' => (object) ['$eq' => 1]]];
         yield 'field query operator' => [['foo' => new EqOperator(1)]];
         yield 'query operator' => [[new CommentOperator('foo'), 'foo' => 1], 2];
+        yield 'numeric field with array' => [[1 => [2, 3]]];
+        yield 'numeric field with operator' => [[1 => ['$eq' => 2]]];
     }
 
     public function testFieldQueryList(): void
     {
         $queryObject = QueryObject::create(
-            foo: [new GtOperator(1), new LtOperator(5)],
+            ['foo' => [new GtOperator(1), new LtOperator(5)]],
```

Comment:
In an earlier comment, @alcaeus wrote:

> LGTM provided that the named argument syntax continues to work

Are you still testing with named arg syntax elsewhere?

Reply:
Yes, but not for this internal constructor. The variadic named arguments is supported by `MongoDB\Builder\Query::query(...$queries)`

Reply:
Tests on `MongoDB\Builder\Query::query(...$queries)` are only in the integration tests (BuilderEncoderTests and various operator tests).